From 39c9fda4df4c58c8d300df7be770cd123e43b938 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 5 Nov 2015 11:33:08 +0800
Subject: [PATCH 556/602] MA-7217 Make fsl-sdcard-partition.sh to be able to
 program sparse image

Use simg2img to translate the sparse image into raw image
dd the raw image into ext4 system partition

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 common/tools/fsl-sdcard-partition-f2fs.sh | 8 +++++++-
 common/tools/fsl-sdcard-partition.sh      | 8 +++++++-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/common/tools/fsl-sdcard-partition-f2fs.sh b/common/tools/fsl-sdcard-partition-f2fs.sh
index ec9dbe4..fed808d 100755
--- a/common/tools/fsl-sdcard-partition-f2fs.sh
+++ b/common/tools/fsl-sdcard-partition-f2fs.sh
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# android-tools-fsutils should be installed as
+# "sudo apt-get install android-tools-fsutils"
+
 # partition size in MB
 BOOTLOAD_RESERVE=8
 BOOT_ROM_SIZE=16
@@ -36,6 +39,7 @@ not_format_fs=0
 bootloader_file="u-boot.imx"
 bootimage_file="boot.img"
 systemimage_file="system.img"
+systemimage_raw_file="system_raw.img"
 recoveryimage_file="recovery.img"
 while [ "$moreoptions" = 1 -a $# -gt 0 ]; do
 	case $1 in
@@ -103,7 +107,9 @@ if [ "${flash_images}" -eq "1" ]; then
     dd if=/dev/zero of=${node} bs=1k seek=384 count=129
     dd if=${bootimage_file} of=${node}${part}1
     dd if=${recoveryimage_file} of=${node}${part}2
-    dd if=${systemimage_file} of=${node}${part}5
+    simg2img ${systemimage_file} ${systemimage_raw_file}
+    dd if=${systemimage_raw_file} of=${node}${part}5
+    rm ${systemimage_raw_file}
     dd if=${bootloader_file} of=${node} bs=1k seek=1
     sync
 fi
diff --git a/common/tools/fsl-sdcard-partition.sh b/common/tools/fsl-sdcard-partition.sh
index 08e0050..07e71c6 100755
--- a/common/tools/fsl-sdcard-partition.sh
+++ b/common/tools/fsl-sdcard-partition.sh
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# android-tools-fsutils should be installed as
+# "sudo apt-get install android-tools-fsutils"
+
 # partition size in MB
 BOOTLOAD_RESERVE=8
 BOOT_ROM_SIZE=16
@@ -36,6 +39,7 @@ not_format_fs=0
 bootloader_file="u-boot.imx"
 bootimage_file="boot.img"
 systemimage_file="system.img"
+systemimage_raw_file="system_raw.img"
 recoveryimage_file="recovery.img"
 while [ "$moreoptions" = 1 -a $# -gt 0 ]; do
 	case $1 in
@@ -104,7 +108,9 @@ if [ "${flash_images}" -eq "1" ]; then
     dd if=${bootloader_file} of=${node} bs=1k seek=1
     dd if=${bootimage_file} of=${node}${part}1
     dd if=${recoveryimage_file} of=${node}${part}2
-    dd if=${systemimage_file} of=${node}${part}5
+    simg2img ${systemimage_file} ${systemimage_raw_file}
+    dd if=${systemimage_raw_file} of=${node}${part}5
+    rm ${systemimage_raw_file}
     sync
 fi
 }
-- 
1.8.0

